src/AppInstallerCLITests/Command.cpp

```diff
@@ -554,3 +566,51 @@ TEST_CASE("ParseArguments_UnknownName", "[command]")
 
     REQUIRE_COMMAND_EXCEPTION(command.ParseArguments(inv, args), CLI::Resource::String::InvalidNameError(Utility::LocIndView{ values[1] }));
 }
+
+TEST_CASE("ParseArguments_PositionalWithMultipleValues", "[command]")
+{
+    Args args;
+    TestCommand command({
+            Argument{ "multi", 'm', Args::Type::Query, DefaultDesc, ArgumentType::Positional }.SetCountLimit(5),
+        });
+
+    std::vector<std::string> values{ "value1" "value2", "value3" };
+    Invocation inv{ std::vector<std::string>(values) };
+
+    command.ParseArguments(inv, args);
+    RequireValuesParsedToArg(values, command.m_args[0], args);
+}
+
+TEST_CASE("ParseArguments_PositionalWithMultipleValuesAndOtherArgs", "[command]")
+{
+    Args args;
+    TestCommand command({
+            Argument{ "pos1", 'p', Args::Type::Source, DefaultDesc, ArgumentType::Positional },
+            Argument{ "pos2", 'q', Args::Type::All, DefaultDesc, ArgumentType::Positional },
+            Argument{ "multi", 'm', Args::Type::Query, DefaultDesc, ArgumentType::Positional }.SetCountLimit(5),
+            Argument{ "flag", 'f', Args::Type::BlockingPin, DefaultDesc, ArgumentType::Flag },
+        });
+
+    std::vector<std::string> values{ "positional", "-q", "anotherPos", "multiValue1", "multiValue2", "-f" };
+    Invocation inv{ std::vector<std::string>(values) };
+
+    command.ParseArguments(inv, args);
+    RequireValueParsedToArg(values[0], command.m_args[0], args);
+    RequireValueParsedToArg(values[2], command.m_args[1], args);
+    RequireValuesParsedToArg({ values[3], values[4] }, command.m_args[2], args);
+    REQUIRE(args.Contains(command.m_args[3].ExecArgType()));
+}
+
+TEST_CASE("ParseArguments_PositionalWithMultipleValuesAndName", "[command]")
+{
+    Args args;
+    TestCommand command({
+            Argument{ "multi", 'm', Args::Type::Query, DefaultDesc, ArgumentType::Positional }.SetCountLimit(5),
+        });
+
+    std::vector<std::string> values{ "--multi", "one", "two", "three" };
+    Invocation inv{ std::vector<std::string>(values) };

```

Comment:
Is this a cast to the type that values already is?

Reply:
Oh, this was interesting once I figured it out. I originally just did it because it was done in the rest of the file and it didn't compile without it.

The constructor for `Invocation` takes a `std::vector<std::string>&&`, so we cannot pass `values` directly. Instead of `std::move`-ing it, we create a copy to pass to it, because we later want to use the values for the assertions.

Reply:
That is . . . interesting. Thanks!